src/sidebar/media-embedder.js

```diff
@@ -25,11 +25,88 @@ function iframe(src) {
   return iframe_;
 }
 
+/**
+ * Return timeValue as a value in seconds, supporting `t` param's optional
+ * '\dh\dm\ds' format. If `timeValue` is numeric (only),
+ * it's assumed to be seconds and is left alone.
+ *
+ * @param {string} timeValue - value of `t` or `start` param in YouTube URL
+ * @returns {string} timeValue in seconds
+ * @example
+ * formatYouTubeTime('5m'); // returns '300'
+ * formatYouTubeTime('20m10s'); // returns '1210'
+ * formatYouTubeTime('1h1s'); // returns '3601'
+ * formatYouTubeTime('10'); // returns '10'
+ **/
+function formatYouTubeTime (timeValue) {
```

Comment:
Perhaps `parseTimeString` might be a better name for this function as it takes a formatted input and extracts data from it rather than returning a pretty representation of some data as a "formatXXX" function usually might do.